channeldb/kvdb/etcd/stm.go

```diff
@@ -403,12 +403,23 @@ func (s *stm) Get(key string) ([]byte, error) {
 	// the prefetch set.
 	if getValue, ok := s.prefetch[key]; ok {
 		delete(s.prefetch, key)
-		s.rset[key] = getValue
+
+		// Use the prefetched value only if it is for
+		// an existing key.
+		if getValue.rev != 0 {
+			s.rset[key] = getValue
+		}
 	}
 
 	// Return value if alread in read set.
-	if getVal, ok := s.rset[key]; ok {
-		return []byte(getVal.val), nil
+	if getValue, ok := s.rset[key]; ok {
+		// Return the value if the rset contains
+		// an existing key.
+		if getValue.rev != 0 {
+			return []byte(getValue.val), nil
+		} else {
+			return nil, nil
```

Comment:
is it possible for a bucket to have a non-zero rev?

Reply:
all values committed to the db have non-zero mod revision